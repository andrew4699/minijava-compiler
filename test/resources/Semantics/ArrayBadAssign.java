class Factorial{
	public static void main(String[] a){
		System.out.println(new A().ComputeFac(10));
	}
}

class A {
	int[] a;
	int[] b;
  boolean c;

	public int ComputeFac(int num){
		int num_aux ;
		num_aux = 5;

    a = new int[num_aux];
    a[0] = 1;
    a[1] = true;

    b[0] = 2;

		c = new int[30];
    return num_aux;
	}
}
