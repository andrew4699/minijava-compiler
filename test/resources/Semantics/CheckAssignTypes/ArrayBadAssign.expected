[CheckAssignTypesVisitor - 18]: Trying to assign boolean to int
[CheckAssignTypesVisitor - 22]: Trying to assign int[] to boolean
